### PR TITLE
implemented flagging component front end

### PR DESCRIPTION
**Pull request by bangpham**

### CODE REVIEW

**Comment by bangpham** [comment]
not sure what this is in package-lock.json. In case anything breaks, this is the conflicts I resolved.

<<<<<<< bangflaglisting
    "handlebars": {
      "version": "1.3.0",
      "resolved": "https://registry.npmjs.org/handlebars/-/handlebars-1.3.0.tgz",
      "integrity": "sha1-npsTCpPjiUkTItl1zz7BgYw3zjQ=",
      "dev": true,
      "requires": {
        "optimist": "0.3.7",
        "uglify-js": "2.3.6"
      },
      "dependencies": {
        "async": {
          "version": "0.2.10",
          "resolved": "https://registry.npmjs.org/async/-/async-0.2.10.tgz",
          "integrity": "sha1-trvgsGdLnXGXCMo43owjfLUmw9E=",
          "dev": true,
          "optional": true
        },
        "source-map": {
          "version": "0.1.43",
          "resolved": "https://registry.npmjs.org/source-map/-/source-map-0.1.43.tgz",
          "integrity": "sha1-wkvBRspRfBRx9drL4lcbK3+eM0Y=",
          "dev": true,
          "optional": true,
          "requires": {
            "amdefine": "1.0.1"
          }
        },
        "uglify-js": {
          "version": "2.3.6",
          "resolved": "https://registry.npmjs.org/uglify-js/-/uglify-js-2.3.6.tgz",
          "integrity": "sha1-+gmEdwtCi3qbKoBY9GNV0U/vIRo=",
          "dev": true,
          "optional": true,
          "requires": {
            "async": "0.2.10",
            "optimist": "0.3.7",
            "source-map": "0.1.43"
          }
        }
      }
    },
=======
>>>>>>> master

**Review by matthewkohlhaas — APPROVED** [review]
I went ahead and made the necessary fixes so that Angular Material is used instead of Bootstrap.

**Comment by matthewkohlhaas** [comment]
Don't merge this until the backend branch for this is merged.

File: src/services/flag.service.ts
Comment (matthewkohlhaas) [inline src/services/flag.service.ts]: Note: You will have to change the route here if you change it on the backend as requested in the backend PR.

**Comment by matthewkohlhaas** [comment]
WARNING: Don't merge this until the backend branch for this is merged.

**Pull request merged**
